async-indexing/src/main/java/zone/cogni/asquare/service/index/IndexConfigProvider.java

```diff
@@ -1,134 +1,134 @@
+import zone.cogni.libs.sparqlservice.SparqlService;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+@Builder
+@Getter
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class IndexConfigProvider {
+
+  private final RdfStoreService rdfStoreService;
+  private final Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier;
+  private final ElasticStore elasticStore;
+  @Nullable
+  private final Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier;
+  /**
+   * Post index interceptor consuming the resulting index document and receiving the Type of the object.
+   * This allows to tweak the index document before it's sent to elasticsearch.
+   */
+  @Nullable
+  private final BiConsumer<ObjectNode, String> postIndexInterceptor;
+  @Getter(lazy = true)
+  private final SparqlService sparqlService = new RdfStoreSparqlService(rdfStoreService); //init will be done on first get call
+
+  /**
+   * @deprecated Use builder instead.
+   */
+  @Deprecated
+  public IndexConfigProvider(RdfStoreService rdfStoreService,
+                             Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
+                             Elasticsearch7Store elasticStore,
+                             Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier) {
+    this(rdfStoreService, applicationProfileSupplier, elasticStore, facetConversionSupplier, null);
+  }
+
+  /**
+   * @deprecated Use builder instead.
+   */
+  @Deprecated
+  public IndexConfigProvider(RdfStoreService rdfStoreService,
+                             Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
+                             Elasticsearch7Store elasticStore) {
+    this(rdfStoreService, applicationProfileSupplier, elasticStore, null, null);
+  }
+
+  /**
+   * @deprecated Use builder instead.
+   */
+  @Deprecated
+  public IndexConfigProvider(SparqlService sparqlService,
+                             Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
+                             Elasticsearch7Store elasticStore) {
+    this(new SparqlRdfStoreService(sparqlService), applicationProfileSupplier, elasticStore, null, null);
+  }
+
+  /**
+   * @deprecated Use builder instead.
+   */
+  @Deprecated
+  public IndexConfigProvider(SparqlService sparqlService,
+                             Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
+                             Elasticsearch7Store elasticStore,
+                             Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier) {
+    this(new SparqlRdfStoreService(sparqlService), applicationProfileSupplier, elasticStore, facetConversionSupplier, null);
+  }
+
+  /**
+   * @deprecated Use {@link RdfStoreSparqlService}
+   */
+  @Deprecated

```

Comment:
This class seems to be only used in https://github.com/cognizone/asquare/blob/c3c805e0f41046b1c2a027dcdaa8a352715add03/async-indexing/src/main/java/zone/cogni/asquare/service/index/GraphIndexService.java#L61

Should not we make it at least package default (if not delete it)?